thunder/core/decorators.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+
+def avoid_torch_nccl_record_streams(func):
```

Comment:
Is there a better pattern for patching env variables in Python?

Reply:
I think It's better to set this env variable once at the import time of Thunder.

The current approach is not reliable because processes do not share the environment and execute functions at different speeds. I think it's very likely that one process finishes faster and unsets this variable while others are still attempting to read it.

Reply:
Judging from here https://github.com/pytorch/pytorch/pull/89880/files
it looks like it's the only way for now.

Given the bump in perf I'd look at merging this one and try to advocate for a proper flag in parallel.